main.go

```diff
@@ -79,19 +189,75 @@ func (b *Backend) healthCheck() {
 			fallthrough
 		case err == nil && resp.StatusCode == http.StatusOK:
 			resp.Body.Close()
-			if b.logging {
-				fmt.Printf("%s is up\n", b.endpoint)
+			var downtimeEnd time.Time
+			if (b.DowntimeStart != time.Time{}) {
+				now := time.Now().UTC()
+				downtime := now.Sub(b.DowntimeStart)
+				b.updateDowntime(downtime)
+				downtimeEnd = now
+			}
+			if globalLogFlag && !b.up && (b.Stats.UpSince != time.Time{}) {
+				log(logMessage{Endpoint: b.endpoint, Status: "up", DowntimeStart: b.DowntimeStart, DowntimeEnd: downtimeEnd})
 			}
+			b.Stats.UpSince = time.Now()
 			b.up = true
+			b.DowntimeStart = time.Time{}
 		default:
-			if b.logging {
-				fmt.Printf("%s is down : %s\n", b.endpoint, err)
+			var errMsg string
+			if err != nil {
+				errMsg = err.Error()
+			} else {
+				errMsg = resp.Status
+			}
+			if globalLogFlag && (b.up || (b.Stats.UpSince == time.Time{})) {
+				log(logMessage{Endpoint: b.endpoint, Status: "down", Error: errMsg})
```

Comment:
Make this Status a `BackendStatus` type, which evaluates into `"down"` when marshalled by JSON